src/QsCompiler/Tests.Compiler/TestUtils/TestUtils.fs

```diff
@@ -276,3 +280,206 @@ let testExpr (str, succExp, resExp, diagsExp) =
+    |> Seq.find (fun x -> x.Key.Name = name)
+    |> (fun x -> x.Value)
+
+let getCallablesWithSuffix compilation ns (suffix: string) =
+    compilation.Namespaces
+    |> Seq.filter (fun x -> x.Name = ns)
+    |> GlobalCallableResolutions
+    |> Seq.filter (fun x -> x.Key.Name.EndsWith suffix)
+    |> Seq.map (fun x -> x.Value)
+
+let private _checkIfStringIsCall (name: QsQualifiedName) input =
+    let call = sprintf @"(%s\.)?%s" <| Regex.Escape name.Namespace <| Regex.Escape name.Name
+    let typeArgs = @"(<\s*([^<]*[^<\s])\s*>)?" // Does not support nested type args
+    let args = @"\(\s*(.*[^\s])?\s*\)"
+    let regex = sprintf @"^\s*%s\s*%s\s*%s;$" call typeArgs args
+
+    let regexMatch = Regex.Match(input, regex)
+
+    if regexMatch.Success then
+        (true, regexMatch.Groups.[3].Value, regexMatch.Groups.[4].Value)
+    else
+        (false, "", "")
+
+let identifyGeneratedByCalls generatedCallables calls =
+    let mutable callables =
+        generatedCallables |> Seq.map (fun x -> x, x |> (getBodyFromCallable >> getLinesFromSpecialization))
+
+    let hasCall callable call =
+        let (_, lines: string []) = callable
+        call |> Seq.forall (fun (i, name) -> _checkIfStringIsCall name lines.[i] |> (fun (x, _, _) -> x))
+
+    Assert.True(Seq.length callables = Seq.length calls) // This should be true if this method is called correctly
+
+    let mutable rtrn = Seq.empty
+
+    let removeAt i lst =
+        Seq.append <| Seq.take i lst <| Seq.skip (i + 1) lst
+
+    for call in calls do
+        callables
+        |> Seq.tryFindIndex (fun callSig -> hasCall callSig call)
+        |> (fun x ->
+            Assert.True(x <> None, "Did not find expected generated content")
+            rtrn <- Seq.append rtrn [ Seq.item x.Value callables ]
+            callables <- removeAt x.Value callables)
+
+    rtrn |> Seq.map (fun (x, _) -> x)
+
+let identifyCallablesBySignature generatedCallables signatures =
+    let mutable callables =
+        generatedCallables
+        |> Seq.map
+            (fun x ->
+                x,
+                (SyntaxTreeToQsharp.ArgumentTuple(x.ArgumentTuple, (fun x -> SyntaxTreeToQsharp.Default.ToCode(x))),
+                 SyntaxTreeToQsharp.Default.ToCode(x.Signature.ReturnType)))
+
+    Assert.True(Seq.length callables = Seq.length signatures) // This should be true if this method is called correctly
+
+    let mutable rtrn = Seq.empty
+
+    let removeAt i lst =
+        Seq.append <| Seq.take i lst <| Seq.skip (i + 1) lst
+
+    for signature in signatures do
+        callables
+        |> Seq.tryFindIndex (fun callSig -> signature = (snd callSig))
+        |> (fun x ->
+            Assert.True(x <> None, "Did not find expected generated content")
+            rtrn <- Seq.append rtrn [ Seq.item x.Value callables |> fst ]
+            callables <- removeAt x.Value callables)
+
+    rtrn
+
+
+// utils for testing checks and assertions
+
+let checkIfStringIsCall = _checkIfStringIsCall
```

Comment:
nit: Can you update the original `let private _checkIfStringIsCall` instead?

```suggestion
```

Reply:
It is done this way because let bindings are processed in order from top to bottom in F# and I needed the `_checkIfStringIsCall` for the `identifyGeneratedByCalls` function. However the proper `checkIfStringIsCall` function should be in the last grouping of functions. The requirement that functions be defined in a specific order is causing this awkward solution.

Reply:
Alternatively, I have considered that maybe both the `checkIfStringIsCall` and the `identifyGeneratedByCalls` are not that generally useful and should be moved back to the ClassicallyControlled test file.

Reply:
OK. I would personally go with a slightly wonky function grouping that reflects the dependencies accurately, instead of declaring the same function twice with two different names, but it's up to you.

Reply:
I've moved the functions dealing with checking if lines are calls back to the Classically Controlled test module.